web/src/dialog_widget.ts

```diff
@@ -139,6 +142,33 @@ export function launch(conf: WidgetConfig): void {
 
     const $submit_button = $dialog.find(".dialog_submit_button");
 
+    function get_current_values($inputs: JQuery): Record<string, unknown> {
+        const current_values: Record<string, unknown> = {};
+        $inputs.each(function () {
+            const property_name = $(this).attr("name")!;
+            current_values[property_name] = $(this).val();
+        });
+        return current_values;
+    }
+
+    if (conf.update_submit_disabled_state_on_change) {
+        const $inputs = $dialog.find(".modal__content :input");
+
+        const original_values = get_current_values($inputs);
+
+        $submit_button.prop("disabled", true);
+
+        $inputs.on("input", () => {
+            const current_values = get_current_values($inputs);
+
+            if (JSON.stringify(original_values) !== JSON.stringify(current_values)) {
```

Comment:
Hmm, actually, this comparison doesn't make any sense -- JSON stringification is not necessarily stable if there's multiple items in the objects. Changed to `_.isEqual`, which is an actual function for deep equality comparisons of objects.